CMakeLists.txt

```diff
@@ -0,0 +1,148 @@
+#
+# Copyright 2016 Rene Widera, Benjamin Worpitz, Simeon Ehrig
+#
+# This file is part of cupla.
+#
+# cupla is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cupla is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with cupla.
+# If not, see <http://www.gnu.org/licenses/>.
+#
+
+################################################################################
+# Required cmake version.
+################################################################################
+
+cmake_minimum_required(VERSION 3.18.0)
+
+################################################################################
+# Setup project information
+################################################################################
+
+# Find cupla version.
+file(STRINGS "${CMAKE_CURRENT_LIST_DIR}/include/cupla/version.hpp" CUPLA_VERSION_MAJOR_HPP REGEX "#define CUPLA_VERSION_MAJOR ")
+file(STRINGS "${CMAKE_CURRENT_LIST_DIR}/include/cupla/version.hpp" CUPLA_VERSION_MINOR_HPP REGEX "#define CUPLA_VERSION_MINOR ")
+file(STRINGS "${CMAKE_CURRENT_LIST_DIR}/include/cupla/version.hpp" CUPLA_VERSION_PATCH_HPP REGEX "#define CUPLA_VERSION_PATCH ")
+
+string(REGEX MATCH "([0-9]+)" CUPLA_VERSION_MAJOR  ${CUPLA_VERSION_MAJOR_HPP})
+string(REGEX MATCH "([0-9]+)" CUPLA_VERSION_MINOR  ${CUPLA_VERSION_MINOR_HPP})
+string(REGEX MATCH "([0-9]+)" CUPLA_VERSION_PATCH  ${CUPLA_VERSION_PATCH_HPP})
+
+set(PACKAGE_VERSION "${CUPLA_VERSION_MAJOR}.${CUPLA_VERSION_MINOR}.${CUPLA_VERSION_PATCH}")
+
+
+project(cupla VERSION      ${CUPLA_VERSION_MAJOR}.${CUPLA_VERSION_MINOR}.${CUPLA_VERSION_PATCH}
+              DESCRIPTION  "cupla is a simple CUDA like user interface for the platform independent parallel kernel acceleration library alpaka."
+              HOMEPAGE_URL "https://github.com/alpaka-group/cupla"
+              LANGUAGES    CXX)
+include(GNUInstallDirs)
+
+################################################################################
+# cupla options
+################################################################################
+
+option(CUPLA_STREAM_ASYNC_ENABLE "Enable asynchronous streams" ON)
+
+################################################################################
+# setup alpaka
+################################################################################
+
+# the min and max. supported alpaka version is also copied to the cuplaConfig.cmake
+set(_CUPLA_MIN_ALPAKA_VERSION 0.6.0)
+set(_CUPLA_UNSUPPORTED_ALPAKA_VERSION 0.7.0)
```

Comment:
What happens if someone tries this with alpaka 0.5?

Reply:
Should throw an an error at cmake configure time, because cupla 0.3 is not compatible with the old alpaka interface.

Reply:
By the way. I also want to go to CMake 3.19. Then we have support for version ranges. I think, this is quit useful for alpaka. But I'm not sure, if alpaka also need to use CMake 3.19. I have to test this.

Reply:
This should work without alpaka having to change. Otherwise you'd have to manually change the CMake version of every third party library you want to use.

Reply:
I also thought it. My first try failed. I could use an unsupported alpaka version, but it is really possible, that I did a mistake.